header-verifier/tests/header_verifier_blackbox_test.rs

```diff
@@ -1,125 +1,313 @@
+
+        ManagedBuffer::from(&sha256(array))
     }
 }
 
 #[test]
 fn test_deploy() {
     let mut state = HeaderVerifierTestState::new();
-    state.deploy_header_verifier_contract();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
+
+    state.deploy_header_verifier_contract(managed_bls_keys);
+}
+
+#[test]
+fn test_register_esdt_address() {
+    let mut state = HeaderVerifierTestState::new();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
+
+    state.deploy_header_verifier_contract(managed_bls_keys);
+    state.propose_register_esdt_address(ENSHRINE_ADDRESS);
+
+    state
+        .world
+        .query()
+        .to(HEADER_VERIFIER_ADDRESS)
+        .whitebox(header_verifier::contract_obj, |sc| {
+            let esdt_address = sc.esdt_safe_address().get();
+
+            assert_eq!(esdt_address, ENSHRINE_ADDRESS);
+        })
+}
+
+#[test]
+fn test_register_bridge_operation() {
+    let mut state = HeaderVerifierTestState::new();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
+
+    state.deploy_header_verifier_contract(managed_bls_keys);
+
+    let operation_1 = ManagedBuffer::from("operation_1");
+    let operation_2 = ManagedBuffer::from("operation_2");
+    let operation = state.get_mock_operation(vec![&operation_1, &operation_2]);
+
+    state.propose_register_operations(operation);
+
+    state
+        .world
+        .query()
+        .to(HEADER_VERIFIER_ADDRESS)
+        .whitebox(header_verifier::contract_obj, |sc| {
+            assert!(!sc.hash_of_hashes_history().is_empty());
+        });
 }
 
 #[test]
-fn test_register_bridge_ops_wrong_hashes() {
+fn test_remove_executed_hash_caller_not_esdt_address() {
     let mut state = HeaderVerifierTestState::new();
-    state.deploy_header_verifier_contract();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
 
-    let bridge_operations_hash =
-        ManagedBuffer::from("6ee1e00813a74f8293d2c63172c062d38bf780d8811ff63984813a49cd61ff9e");
-    let mock_signature: BlsSignature<StaticApi> = ManagedByteArray::new_from_bytes(
-        b"EIZ2\x05\xf7q\xc7G\x96\x1f\xba0\xe2\xd1\xf5pE\x14\xd7?\xac\xff\x8d\x1a\x0c\x11\x900f5\xfb\xff4\x94\xb8@\xc5^\xc2,exn0\xe3\xf0\n"
+    state.deploy_header_verifier_contract(managed_bls_keys);
+
+    let operation_1 = ManagedBuffer::from("operation_1");
+    let operation_2 = ManagedBuffer::from("operation_2");
+    let operation = state.get_mock_operation(vec![&operation_1, &operation_2]);
+
+    let error_status = ErrorStatus {
+        code: 4,
+        error_message: "Only ESDT Safe contract can call this endpoint",
+    };
+
+    state.propose_register_operations(operation.clone());
+    state.propose_register_esdt_address(ENSHRINE_ADDRESS);
```

Comment:
It is very hard to understand from this test that will return that error. You should add `caller` argument in `propose_remove_execute_hash` function

Reply:
I did that because test setup convenience, I will have to setup another contract for that. If i use it as an argument to the function I have to put it as `TestSCAddress`